prediction_market_agent/agents/microchain_agent/microchain_agent.py

```diff
@@ -3,33 +3,48 @@
 from microchain.functions import Reasoning, Stop
 from prediction_market_agent_tooling.markets.markets import MarketType
 
-from prediction_market_agent.agents.microchain_agent.functions import (
+from prediction_market_agent.agents.microchain_agent.agent_functions import (
+    AGENT_FUNCTIONS,
+)
+from prediction_market_agent.agents.microchain_agent.learning_functions import (
+    LEARNING_FUNCTIONS,
+)
+from prediction_market_agent.agents.microchain_agent.market_functions import (
     MARKET_FUNCTIONS,
-    RememberPastLearnings,
 )
 from prediction_market_agent.agents.microchain_agent.memory import LongTermMemory
+from prediction_market_agent.agents.microchain_agent.memory_functions import (
+    RememberPastLearnings,
+)
 from prediction_market_agent.agents.microchain_agent.omen_functions import (
     OMEN_FUNCTIONS,
 )
+from prediction_market_agent.agents.microchain_agent.prompts import (
+    BOOTSTRAP,
+    TRADING_AGENT_SYSTEM_PROMPT,
+)
 from prediction_market_agent.agents.utils import LongTermMemoryTaskIdentifier
 from prediction_market_agent.utils import APIKeys
 
-SYSTEM_PROMPT = """
-Act as a trader agent in prediction markets to maximise your profit.
 
-Research markets, buy tokens you consider undervalued, and sell tokens that you
-hold and consider overvalued.
+class CustomizedAgent(Agent):
+    """
+    Subclass microchain's agent for any customizations that are needed right away.
+    Anything generally useful should be moved upstream to the microchain repository.
+    """
```

Comment:
I don't follow - is `Agent` intended to be an abstract class? if yes, we should inherit from ABC to make it explicit.

Another question - we only have 1 microchain agent, why do you think it's a good idea to use inheritance here?

Reply:
PR opened here https://github.com/galatolofederico/microchain/pull/11

When it's merged I'll remove `build_initial_messages` in this `CustomizedAgent` class and we can either keep it  plain or remove completely -- no preference on my side atm.